### PR TITLE
fix(spm): add product lines to Package.swift

### PR DESCRIPTION
**Pull request by markemer**
Currently in draft as this fixes one problem but is dependent on the following

- [x] Changing the SPM Repo name
- [x] Fixing the Names in all the Package.swift files 
    - https://github.com/ionic-team/capacitor-plugins/pull/2041
- [x] Fixing the iOS Template
    - #7279
- [x] Clean up some of the parsing assumptions

### CODE REVIEW

**Review by giralte-ionic — APPROVED** [review]

**Review by jcesarmobile — CHANGES_REQUESTED** [review]
This fails to detect deleted plugins, they are kept in the `Package.swift` file, making the app fail.

I think we should go with full file generation (or at least everything after the `// DO NOT MODIFY THIS FILE`) instead of appending the plugins to the existing ones, this way it will generate the file without the uninstalled plugins, otherwise we would need to keep track of installed plugins on every sync and compare with the previous list and update the list, etc.

**Comment by jcesarmobile** [comment]
another idea is to keep the file as is and append the dependencies to the end, I've tried this and it works:

```
package.dependencies.append(.package(name: "CapacitorActionSheet", path: "../capacitor-plugins/action-sheet"))

package.targets.first?.dependencies.append(.product(name: "CapacitorActionSheet", package: "CapacitorActionSheet"))
```

**Comment by markemer** [comment]
Thanks for the bug callout - generating the whole thing from the start might be the answer, or something programmatic.

File: cli/src/util/spm.ts [hunk 1]
```diff
@@ -10,6 +10,15 @@ export interface SwiftPlugin {
   path: string;
 }
 
+export async function checkPackageManager(config: Config): Promise<string> {
```
Comment (Steven0351) [inline cli/src/util/spm.ts:13]: Maybe make this return a `Promise<'SPM' | 'Cocoapods'>` instead of `string`

File: cli/src/util/spm.ts [hunk 2]
```diff
@@ -18,96 +29,62 @@ export async function findPackageSwiftFile(config: Config): Promise<string> {
...
+    let package = Package(
+        name: "CapApp-SPM",
+        platforms: [.iOS(.v13)],
+        products: [
+            .library(
+                name: "CapApp-SPM",
+                targets: ["CapApp-SPM"])
+        ],
+        dependencies: [
+            .package(url: "https://github.com/ionic-team/capacitor-spm.git", branch: "main")`;
```
Comment (jcesarmobile) [inline cli/src/util/spm.ts:62]:
```suggestion
  let packageSwiftText = `// swift-tools-version: 5.9
import PackageDescription

// DO NOT MODIFY THIS FILE - managed by Capacitor CLI commands
let package = Package(
    name: "CapApp-SPM",
    platforms: [.iOS(.v13)],
    products: [
        .library(
            name: "CapApp-SPM",
            targets: ["CapApp-SPM"])
    ],
    dependencies: [
        .package(url: "https://github.com/ionic-team/capacitor-spm.git", branch: "main")`;
```

File: cli/src/util/spm.ts [hunk 3]
```diff
@@ -18,96 +29,62 @@ export async function findPackageSwiftFile(config: Config): Promise<string> {
...
+
+  for (const plugin of plugins) {
+    const relPath = relative(config.ios.nativeXcodeProjDirAbs, plugin.rootPath);
+    packageSwiftText += `,\n            .package(name: "${plugin.ios?.name}", path: "${relPath}")`;
```
Comment (jcesarmobile) [inline cli/src/util/spm.ts:66]:
```suggestion
    packageSwiftText += `,\n        .package(name: "${plugin.ios?.name}", path: "${relPath}")`;
```

File: cli/src/util/spm.ts [hunk 4]
```diff
@@ -18,96 +29,62 @@ export async function findPackageSwiftFile(config: Config): Promise<string> {
...
 
-  return 'Cocoapods';
+  packageSwiftText += `
+       ],
+       targets: [
+          .target(
+                name: "CapApp-SPM",
+                dependencies: [
+                    .product(name: "Capacitor", package: "capacitor-spm"),
+                    .product(name: "Cordova", package: "capacitor-spm")`;
```
Comment (jcesarmobile) [inline cli/src/util/spm.ts:76]:
```suggestion
    ],
    targets: [
        .target(
            name: "CapApp-SPM",
            dependencies: [
                .product(name: "Capacitor", package: "capacitor-spm"),
                .product(name: "Cordova", package: "capacitor-spm")`;
```

File: cli/src/util/spm.ts [hunk 5]
```diff
@@ -18,96 +29,62 @@ export async function findPackageSwiftFile(config: Config): Promise<string> {
...
+
+  for (const plugin of plugins) {
+    packageSwiftText += `,\n                    .product(name: "${plugin.ios?.name}", package: "${plugin.ios?.name}")`;
```
Comment (jcesarmobile) [inline cli/src/util/spm.ts:79]:
```suggestion
    packageSwiftText += `,\n                .product(name: "${plugin.ios?.name}", package: "${plugin.ios?.name}")`;
```

File: cli/src/util/spm.ts [hunk 6]
```diff
@@ -18,96 +29,62 @@ export async function findPackageSwiftFile(config: Config): Promise<string> {
...
+  }
+
+  packageSwiftText += `
+                ]
+            )
+        ]
+      )
+`;
```
Comment (jcesarmobile) [inline cli/src/util/spm.ts:87]:
```suggestion
            ]
        )
    ]
)
`;
```

**Review by jcesarmobile — CHANGES_REQUESTED** [review]
just a few formatting issues on the resulting file

**Pull request merged**